src/docs/development/add-to-app/ios/add-flutter-screen.md

```diff
@@ -6,77 +6,291 @@ description: Learn how to add a single Flutter screen to your existing iOS app.
 
-Make your app delegate implement the `FlutterAppLifeCycleProvider` protocol, for
-example:
+This is not recommended since creating a `FlutterEngine` on-demand could
+introduce a noticeable latency between when the `FlutterViewController` is
+presented and when it will render its first frame. This could, however, be
+sometimes useful if the Flutter screen is rarely shown and there are no good
+heuristics to determine when the Dart VM should be started.
 
-```objective-c
+To let the `FlutterViewController` present without an existing `FlutterEngine`
+simply omit the `FlutterEngine` construction and create the
+`FlutterViewController` without an engine reference.
+
+<ul class="nav nav-tabs" id="no-engine-vc-language" role="tablist">
+  <li class="nav-item">
+    <a class="nav-link active" id="no-engine-vc-objc" href="#no-engine-vc-objc-tab" role="tab" aria-controls="no-engine-vc-objc" aria-selected="true">Objective-C</a>
+  </li>
+  <li class="nav-item">
+    <a class="nav-link" id="no-engine-vc-swift" href="#no-engine-vc-swift-tab" role="tab" aria-controls="no-engine-vc-swift" aria-selected="false">Swift</a>
+  </li>
+</ul>
+
+<div class="tab-content"> {% comment %} No engine FlutterViewController language tab start {% endcomment -%}
+
+<div class="tab-pane active" id="no-engine-vc-objc-tab" role="tabpanel" aria-labelledby="no-engine-vc-objc-tab" markdown="1">
+
+<?code-excerpt "ViewController.m" title?>
+```objectivec
+// Existing code omitted.
+- (void)showFlutter {
+  FlutterViewController *flutterViewController =
+      [[FlutterViewController alloc] initWithProject:nil nibName:nil bundle:nil];
+  [self presentViewController:flutterViewController animated:YES completion:nil];
+}
+@end
+```
+
+</div>
+
+<div class="tab-pane" id="no-engine-vc-swift-tab" role="tabpanel" aria-labelledby="no-engine-vc-swift-tab" markdown="1">
+
+<?code-excerpt "ViewController.swift" title?>
+```swift
+// Existing code omitted.
+func showFlutter() {
+  let flutterViewController = FlutterViewController(project: nil, nibName: nil, bundle: nil)
+  present(flutterViewController, animated: true, completion: nil)
+}
+```
+
+</div>
+
+</div>{% comment %} No engine FlutterViewController language tab end {% endcomment -%}
+
+See [Loading sequence and performance](/docs/development/add-to-app/performance)
+for more explorations on latency and memory usage.
+
+## Using the FlutterAppDelegate
+
+Letting your application's `UIApplicationDelegate` subclass `FlutterAppDelegate`
+is recommended but not required.
+
+The `FlutterAppDelegate` performs functions such as
+
+- Forwarding application callbacks such as [`openURL`](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1623112-application)
+  to plugins such as [local_auth](https://pub.dev/packages/local_auth).
+- Forwarding status bar taps (which can only be detected in the AppDelegate) to
+  Flutter for scroll-to-top behavior.
+
+If your app delegate cannot directly subclass `FlutterAppDelegate`, you can
```

Comment:
s/can/must right?

Reply:
changed to `should`. Must is a bit strong since it's not like the app won't run anymore without it. We can point out the reasoning and let them decide.